File: Bwl.Network.ClientServer.Windows/NetworkAdapter/NetworkAdaptersForm.vb
Imports System.Windows.Forms

Public Class NetworkAdaptersForm

    Public Shared Function SelectAdapterDialog(owner As IWin32Window, selectItemWithKeyword As String) As String
        Dim form As New NetworkAdaptersForm
        form.FillAdapters()
        If selectItemWithKeyword > "" Then
            For i = 0 To form.lbAdapters.Items.Count - 1
                If form.lbAdapters.Items(i).tolower.contains(selectItemWithKeyword.ToLower) Then
                    form.lbAdapters.SelectedIndex = i
                End If
            Next
        End If
        If form.ShowDialog(owner) = DialogResult.OK Then
            Return form.lbAdapters.Text
        End If
        Return ""
    End Function

    Public Sub FillAdapters()
        lbAdapters.Items.Clear()
        Dim adps = NetworkAdapters.GetAdapters
        For Each adp In adps
            lbAdapters.Items.Add(adp.ToString)
        Next
    End Sub

    Private Sub bOK_Click(sender As Object, e As EventArgs) Handles bOK.Click
        If lbAdapters.SelectedIndex >= 0 Then
            DialogResult = DialogResult.OK
            Close()
        End If
    End Sub

    Private Sub bCancel_Click(sender As Object, e As EventArgs) Handles bCancel.Click
        DialogResult = DialogResult.Cancel
        Close()
    End Sub
End Class

File: ClientServer/ClientServer/Bwl.Network.ClientServer.Windows/NetworkAdapter/NetworkAdaptersForm.cs
using System;
using System.Runtime.Versioning;
using System.Windows.Forms;

namespace Bwl.Network.ClientServer.Windows
{

    [SupportedOSPlatform("windows")]
    public partial class NetworkAdaptersForm
    {
        public NetworkAdaptersForm()
        {
            InitializeComponent();
        }

        public static string SelectAdapterDialog(IWin32Window owner, string selectItemWithKeyword)
        {
            var form = new NetworkAdaptersForm();
            form.FillAdapters();
            if (!string.IsNullOrEmpty(selectItemWithKeyword))
            {
                for (int i = 0; i < form.lbAdapters.Items.Count; i++)
                {
                    if (form.lbAdapters.Items[i].ToString().ToLower().Contains(selectItemWithKeyword.ToLower()))
                    {
                        form.lbAdapters.SelectedIndex = i;
                    }
                }
            }
            if (form.ShowDialog(owner) == DialogResult.OK)
            {
                return form.lbAdapters.Text;
            }
            return "";
        }

        public void FillAdapters()
        {
            lbAdapters.Items.Clear();
            NetworkAdapter[] adps = NetworkAdapters.GetAdapters();
            for (int i = 0; i < adps.Length; i++)
            {
                NetworkAdapter? adp = adps[i];
                lbAdapters.Items.Add(adp.ToString());
            }
        }

        private void bOK_Click(object sender, EventArgs e)
        {
            if (lbAdapters.SelectedIndex >= 0)
            {
                DialogResult = DialogResult.OK;
                Close();
            }
        }

        private void bCancel_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }
    }
}